### PR TITLE
Index and prune estimated-time cache

### LINKED ISSUE #3262: EstimatedTimes cache: per-module full-directory enumeration and unbounded growth
**Category:** perf — **Breaking:** no

`Engine/FileSystemModuleEstimatedTimeProvider.cs:33-36` — `GetSubModuleEstimatedTimesAsync` enumerates the entire `%APPDATA%\ModularPipelines\EstimatedTimes` directory (shared across every pipeline the machine ever ran) and filters by name prefix in memory — once per module. O(modules × files-on-disk) syscalls at startup, and nothing ever prunes the directory.

**Fix:** enumerate once and group by module prefix; add age-based pruning.

Part of #3239.

### PR DESCRIPTION
**Pull request by thomhurst**
## Summary
- build one thread-safe estimated-time file index per provider
- group submodule cache files by exact module name
- prune cache entries older than 90 days on index creation
- invalidate the index after saving a submodule estimate
- cover exact matching, reuse, pruning, and invalidation

## Validation
- focused `FileSystemModuleEstimatedTimeProviderTests`: 4 passed
- touched-file whitespace verification: passed
- `ModularPipelines.sln` Release build: 0 errors (existing warnings remain)

Closes #3262

### CODE REVIEW

File: src/ModularPipelines/Engine/FileSystemModuleEstimatedTimeProvider.cs [hunk 1]
```diff
@@ -66,9 +78,94 @@ public async Task<IEnumerable<SubModuleEstimation>> GetSubModuleEstimatedTimesAs
...
+
+    private IReadOnlyDictionary<string, FileInfo[]> GetSubModuleFilesByModule()
+    {
+        var filesByModule = Volatile.Read(ref _subModuleFilesByModule);
+        if (filesByModule is not null)
+        {
+            return filesByModule;
+        }
+
+        lock (_subModuleIndexLock)
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply src/ModularPipelines/Engine/FileSystemModuleEstimatedTimeProvider.cs]: Fixed in 9a19f49779. The index now expires after one minute and is rebuilt under the existing lock, so cache files written by another pipeline process become visible. Added a FakeTimeProvider regression; focused suite passes 5/5.

File: src/ModularPipelines/Engine/FileSystemModuleEstimatedTimeProvider.cs [hunk 2]
```diff
@@ -66,9 +80,110 @@ public async Task<IEnumerable<SubModuleEstimation>> GetSubModuleEstimatedTimesAs
...
+                && now.UtcTicks < _subModuleIndexExpiresAtTicks)
+            {
+                return filesByModule;
+            }
+
+            filesByModule = BuildSubModuleFileIndex(now);
+            Volatile.Write(
+                ref _subModuleIndexExpiresAtTicks,
+                now.Add(IndexRefreshInterval).UtcTicks);
+            Volatile.Write(ref _subModuleFilesByModule, filesByModule);
```
Comment (thomhurst) [inline src/ModularPipelines/Engine/FileSystemModuleEstimatedTimeProvider.cs]: L97-L118: 🟡 risk: files and expiry publish independently; a reader can capture the expired dictionary at L97, then observe the new future expiry at L99 and return that stale index for another minute. Publish one immutable `{ Files, ExpiresAtTicks }` snapshot through a single `Volatile.Write`/`Volatile.Read`.

Comment (thomhurst) [reply src/ModularPipelines/Engine/FileSystemModuleEstimatedTimeProvider.cs]: Fixed in fd917b2fed. The dictionary and expiry now live in one immutable SubModuleFileIndex published through a single Volatile.Write/Read, so readers cannot combine generations. Analyzer verification passes; provider tests pass 5/5.

**Comment by thomhurst** [comment]
Resolved the torn index/expiry publication found by review in fd917b2fed: one immutable snapshot is now published atomically. Focused provider tests pass 5/5; changed-file analyzer verification passes.

**Comment by thomhurst** [comment]
Fixed the remaining refresh-clock edge case in 36e228082e: the immutable cache snapshot now stores a TimeProvider monotonic timestamp, so system-clock rollback cannot extend the one-minute refresh interval. File retention still uses UTC timestamps as intended. Validation: exact changed-file analyzer passed; FileSystemModuleEstimatedTimeProviderTests passed 5/5; simplifier review found no further changes.
